R/modify.R

```diff
@@ -93,41 +93,18 @@ modify <- function(.x, .f, ...) {
 modify.default <- function(.x, .f, ...) {
   .f <- as_mapper(.f, ...)
 
-  for (i in seq_along(.x)) {
-    list_slice2(.x, i) <- .f(.x[[i]], ...)
+  if (vec_is_list(.x) || is.data.frame(.x)) {
+    out <- map(vec_proxy(.x), .f, ...)
+    vec_restore(out, .x)
+  } else if (vec_is(.x)) {
+    map_vec(.x, .f, ..., .ptype = .x)
+  } else if (is.null(.x) || is.list(.x)) {
```

Comment:
It's a bit strange to go through that path for `.x = NULL`. Maybe branch into `return(NULL)`?

Reply:
I had this originally, but it also felt weird — it is nice that `NULL` just falls out as a special case of a zero length list.

Reply:
FWIW I also prefer the explicit `is.null(x)` branch

Reply:
I'm going to leave it with list because (a) it's not that import and (b) it needs the same recycling behaviour as a zero-length list in `modify2()`.